manticore/core/smtlib/visitors.py

```diff
@@ -378,12 +379,12 @@ def visit_BitVecZeroExtend(self, expression, *operands):
             return expression
 
     def visit_BitVecITE(self, expression, *operands):
-        # FIXME Enable some taint propagating optimization
-        if isinstance(expression.operands[0], Constant) and not expression.operands[0].taint:
+        if isinstance(expression.operands[0], Constant):
             if expression.operands[0].value:
                 result = expression.operands[1]
             else:
                 result = expression.operands[2]
+            result._taint |= expression.operands[0].taint
```

Comment:
Idk about the expressions being mutable. Maybe we can just do it for _taint_? But there are caches and other stuff that expects the expressions to be unmutable. This may be happening in other places anyway.

Reply:
> Idk about the expressions being mutable.
I see your point.  I will revise.